src/block_disagg/block_disagg_open.c

```diff
@@ -0,0 +1,222 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __wt_block_disagg_manager_create --
+ *     Create a file - it's a bit of a game with a new block manager. The file is created when
+ *     adding a new table to the metadata, before a btree handle is open. The block storage manager
+ *     is generally created when the btree handle is opened. The caller of this will need to check
+ *     for and instantiate a storage source.
+ */
+int
+__wt_block_disagg_manager_create(
+  WT_SESSION_IMPL *session, WT_BUCKET_STORAGE *bstorage, const char *filename)
+{
+#if 1
+    WT_UNUSED(session);
+    WT_UNUSED(bstorage);
+    WT_UNUSED(filename);
+
+    /*
+     * The default block manager creates the physical underlying file here and writes an initial
+     * block into it. At the moment we don't need to do that for our special storage source - it's
+     * going to magically create the file on first access and doesn't have a block manager provided
+     * special leading descriptor block.
+     */
+    return (0);
+#else
+    WT_DECL_ITEM(tmp);
+    WT_DECL_RET;
+    WT_FH *fh;
+    int suffix;
+    bool exists;
+
+    /*
+     * Create the underlying file and open a handle.
+     *
+     * Since WiredTiger schema operations are (currently) non-transactional, it's possible to see a
+     * partially-created file left from a previous create. Further, there's nothing to prevent users
+     * from creating files in our space. Move any existing files out of the way and complain.
+     */
+    for (;;) {
+        ret = bstorage->file_system->fs_open_file(bstorage->file_system, &session->iface, filename,
+          WT_FS_OPEN_FILE_TYPE_DATA, WT_FS_OPEN_CREATE | WT_FS_OPEN_DURABLE | WT_FS_OPEN_EXCLUSIVE,
+          &fh);
+        if (ret == 0)
+            break;
+        WT_ERR_TEST(ret != EEXIST, ret, false);
+    }
+
+    /* Write out the file's meta-data. */
+    ret = __wt_desc_write(session, fh, allocsize);
+
+    /* Close the file handle. */
+    WT_TRET(__wt_close(session, &fh));
+
+    /* Undo any create on error. */
+    if (ret != 0)
+        WT_TRET(__wt_fs_remove(session, filename, false, false));
+
+err:
+    __wt_scr_free(session, &tmp);
+
+    return (ret);
```

Comment:
Can you add a FIXME to clean up this unused code?

Reply:
I did one better than that -- it's removed! Thanks for spotting this.